### PR TITLE
ValueRef Packing

### PR DESCRIPTION
**Pull request by geoffthemedio**
constexpr and data packing

### CODE REVIEW

File: universe/Planet.h [hunk 1]
```diff
@@ -195,12 +195,39 @@ class FO_COMMON_API Planet final : public UniverseObject {
     template <typename T> friend void boost::python::detail::value_destroyer<false>::execute(T const volatile* p);
 
     static constexpr auto planet_meter_types = []() {
+#if defined(__cpp_using_enum)
         using enum MeterType;
+#else
+        static constexpr auto METER_POPULATION = MeterType::METER_POPULATION;
```
Comment (o01eg) [inline universe/Planet.h]: Can we use boost.preprocessor loop here?

Comment (geoffthemedio) [reply universe/Planet.h]: could do it with standard macros, like most of the enums are defined with...

or could just put all the `MeterType::` back in the various uses below these `using enum` lines.

or could use compilers that actually support C++20 features.

probly I'll just do like `using MT = MeterType` to get most of the shortening benefit.

**Comment by geoffthemedio** [comment]
@o01eg any idea why https://github.com/freeorion/freeorion/actions/runs/12058623847/job/33644613648 fails?

**Comment by o01eg** [comment]
Oh, it seemed Ubuntu Lunar has reached end of life. We use it to test gcc-12.

**Comment by geoffthemedio** [comment]
> gcc-12.

Can you update it to Ubuntu 24.04 and whatever recent GCC version(s), or should it just be removed?

I had a quick look at the relevant seeming scripts but there's too much magic incantation going on for me to fix it.

**Pull request merged**